src/renderer/dx/DxRenderer.cpp

```diff
@@ -1060,6 +1050,16 @@ void DxEngine::_InvalidOr(RECT rc) noexcept
 // - S_OK on success, E_PENDING to indicate a retry or a relevant DirectX error
 [[nodiscard]] HRESULT DxEngine::Present() noexcept
 {
+    if (_retroTerminalEffects)

```

Comment:
Awesome! I bet this makes io feel a little faster since we're not doing computation under console-lock.